packages/grid/_modules_/grid/models/gridOptions.tsx

```diff
@@ -55,6 +55,11 @@ export interface GridOptions {
    * @default false
    */
   checkboxSelection?: boolean;
+  /**
+   * To be used in combination with `checkboxSelection`. If `true`, `Select All` selects only the rows on the current page.
```

Comment:
I think that we should revert the sentence, the to be used seems secondary information.

```suggestion
   * If `true`, the "Select All" header checkbox selects only the rows on the current page. To be used in combination with `checkboxSelection`. 
```